disassemblers/ofrak_cached_disassembly/ofrak_cached_disassembly/components/cached_disassembly_unpacker.py

```diff
@@ -0,0 +1,298 @@
+                )
+                if backend_relative_va == relative_va and backend_cr.size == code_region.size:
+                    code_region.resource.add_view(
+                        backend_cr
+                    )  # TODO: https://github.com/redballoonsecurity/ofrak/issues/537
+        await resource.save()
+
+
+class CachedCodeRegionUnpacker(CodeRegionUnpacker):
+    def __init__(
+        self,
+        resource_factory: ResourceFactory,
+        data_service: DataServiceInterface,
+        resource_service: ResourceServiceInterface,
+        analysis_store: CachedAnalysisStore,
+        component_locator: ComponentLocatorInterface,
+    ):
+        super().__init__(resource_factory, data_service, resource_service, component_locator)
+        self.analysis_store = analysis_store
+
+    async def unpack(self, resource: Resource, config: None):
+        program_r = await resource.get_only_ancestor(ResourceFilter.with_tags(CachedAnalysis))
+        analysis = self.analysis_store.get_analysis(program_r.get_id())
+        if analysis["metadata"]["backend"] == "ghidra":
+            await resource.run(CachedCodeRegionModifier)
+        code_region_view = await resource.view_as(CodeRegion)
+        func_keys = analysis[f"seg_{code_region_view.virtual_address}"]["children"]
+        for func_key in func_keys:
+            complex_block = analysis[func_key]
+            cb = ComplexBlock(
+                virtual_address=complex_block["virtual_address"],
+                size=complex_block["size"],
+                name=complex_block["name"],
+            )
+            await code_region_view.create_child_region(cb)
+
+
+class CachedComplexBlockUnpacker(ComplexBlockUnpacker):
+    def __init__(
+        self,
+        resource_factory: ResourceFactory,
+        data_service: DataServiceInterface,
+        resource_service: ResourceServiceInterface,
+        analysis_store: CachedAnalysisStore,
+        component_locator: ComponentLocatorInterface,
+    ):
+        super().__init__(resource_factory, data_service, resource_service, component_locator)
+        self.analysis_store = analysis_store
+
+    async def unpack(self, resource: Resource, config: None):
+        program_r = await resource.get_only_ancestor(ResourceFilter.with_tags(CachedAnalysis))
+        analysis = self.analysis_store.get_analysis(program_r.get_id())
+        program_attributes = self.analysis_store.get_program_attributes(program_r.get_id())
+
+        cb_view = await resource.view_as(ComplexBlock)
+        child_keys = analysis[f"func_{cb_view.virtual_address}"]["children"]
+        for children in child_keys:
+            if children.startswith("bb"):
+                basic_block = analysis[children]
+                mode = InstructionSetMode.NONE
+                if basic_block["mode"] == "thumb":
+                    mode = InstructionSetMode.THUMB
+                elif basic_block["mode"] == "vle":
+                    mode = InstructionSetMode.VLE
```

Comment:
```suggestion
                if "mode" in basic_block:
                    mode = InstructionSetMode[basic_block["mode"].upper()]
```

I think your code is easier to understand, but this version won't need to be changed if we add more instruction set modes in the future.